traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/adapters/ImageCarouselRecyclerViewAdapter.java

```diff
@@ -0,0 +1,64 @@
+package com.guestlogix.traveleruikit.adapters;
+
+import android.content.res.Resources;
+import android.graphics.Bitmap;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import androidx.recyclerview.widget.RecyclerView;
+import com.guestlogix.traveleruikit.R;
+import com.guestlogix.traveleruikit.tools.AssetManager;
+import com.guestlogix.traveleruikit.tools.image.ImageLoader;
+
+import java.net.URL;
+import java.util.List;
+
+public class ImageCarouselRecyclerViewAdapter extends RecyclerView.Adapter<ImageCarouselRecyclerViewAdapter.ViewHolder> {
+    private List<URL> data;
+
+    @Override
+    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_image_carousel, parent, false);
+        return new ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(ViewHolder holder, int position) {
+        Resources resources = holder.imageView.getResources();
+        AssetManager.getInstance().loadImage(data.get(position),
+                (int) resources.getDimension(R.dimen.thumbnail_width),
+                (int) resources.getDimension(R.dimen.thumbnail_height),
+                holder.imageView.getId(),
+                new ImageLoader.ImageLoaderCallback() {
+                    @Override
+                    public void onBitmapLoaded(Bitmap bitmap) {
+                        holder.imageView.setImageBitmap(bitmap);
+                    }
+
+                    @Override
+                    public void onError() {
+                        holder.imageView.setImageResource(R.color.colorPrimary);
+                    }
+                });
+    }
+
+    @Override
+    public int getItemCount() {
+        return (null != data) ? data.size() : 0;
+    }
+
+    public void setData(List<URL> data) {
+        this.data = data;
+        notifyDataSetChanged();
```

Comment:
should not call notify...its the callers responsibility to call notify when they are ready

Reply:
in fact I think its best to make this property immutable, well make it mutable if we need it

Reply:
changed data to immutable